### ISSUE TITLE
No auto reconnect to wireless after resuming device

### ISSUE THREAD

**Issue by PhieF** [koreader/koreader#3888]
* KOReader version: v2015.11-1666-gf12f312_2018-04-15
* Device: Kobo Aura 2

#### Issue
No auto reconnect to wireless after resuming device
#### Steps to reproduce

Just turn off the screen come back few secondes later, wifi isn't there anymore

**Comment by cramoisi** [comment]
@PhieF sorry to be obvious but was the option "automatically restore wi-fi connection after resume" enabled  ? :-)

**Comment by PhieF** [comment]
Yep hahaha, no problem

**Comment by cramoisi** [comment]
Oups, it's more interesting than that : After the wake-up, I selected my cloud storage and KOReader ask me to reconnect the wifi. I said no and write my comment (my ereader was open on the table during this time). But, after that, I picked it up again, selected again the storage and I was in, so the wifi somehow re-enabled itself during this time. Perhaps it takes more than 5 secondes to reconnect ? Or a false negative on the connection ? @PhieF what about you ?

**Comment by NiLuJe** [comment]
Since I tweaked the WiFi scripts very recently, you might also want to check that they work, *period*, on your device:

Forgetting about suspend/resume, do toggling WiFi from KOReader's menu actually work?

**Comment by PhieF** [comment]
Just to answer your question above, even after some time wifi wasn't
connected (Yesterday version)

**Comment by NiLuJe** [comment]
IIRC, the changes are in since 1662.

I'll see how it behaves on my H2O.

Extra question, if any of you have shell access: what's the output from ```ntx_hwconfig -s -p /dev/mmcblk0 Wifi```?

**Comment by NiLuJe** [comment]
Okay, after toggling WiFi a few times from KOReader, I managed to moderately break it:

Entering the "Network >" menu takes forever (I'm guess it's waiting for something to timeout).

Issue started after getting a couple of apparently hung udhcpc scripts:

```
 1481 root       0:00 {default.script} /bin/sh /etc/udhcpc.d/default.script deconfig
 1486 root       0:00 {default.script} /bin/sh /etc/udhcpc.d/default.script bound
```

**Comment by NiLuJe** [comment]
.... aaaaand one more toggle over the CLI led to a deadlock after trying to re-enable WiFi in the UI. Yay?

**Comment by NiLuJe** [comment]
Okay, definitely seems to be some sort of race, and the sleep change re-triggers it, will revert.

**Comment by NiLuJe** [comment]
Reverting seemed to do the trick, toggled WiFi 6 times without an issue, while it deadlocked on the second try earlier ;).

**Comment by NiLuJe** [comment]
Ah, right.

Well, at least you got me to fix *a* bug, even if it's not exactly this one ^^.

**Comment by cramoisi** [comment]
@PhieF : have you any connexion problem within KSM or nickel ? as for myself, the wifi doesn't work great inside koreader if I don't use it and set it first inside nickel and doesn't work at all if I don't use KSM to launch koreader (aka : if I don't launch nickel and set the wi-fi a first time before going full koreader)

**Comment by Frenzie** [comment]
Sorry, it autoclosed when I merged the other fix and I didn't realize this was apparently a different issue.

@NiLuJe 
> Extra question, if any of you have shell access: what's the output from ntx_hwconfig -s -p /dev/mmcblk0 Wifi 2>/dev/null?

While shell access is very much preferable on general principle, everyone's got it through the Terminal plugin (by @Hzj-jie iirc). ;-)

**Comment by pazos** [comment]
I think this ticket is out-of-date. Anyways it seems a duplicate of #2274.

Closing!
